### PR TITLE
[TTree][NFC] Refurbish a bit the documentation of TTree

### PR DESCRIPTION
**Pull request by dpiparo**

### CODE REVIEW

File: tree/tree/src/TTree.cxx [hunk 1]
```diff
@@ -11,50 +11,55 @@
...
-  to only hold floating-point numbers; a TTree on the other hand can hold all kind
-  of data, such as objects or arrays in addition to all the simple types.
+  In order to store columnar datasets, ROOT provides the TTree, TChain,
+  TNtuple and TNtupleD classes.
+  The TTree class represents a columnar dataset. Any C++ type can be stored in the
+  columns. The TTree has allowed to store about **1 EB** of data coming from the LHC alone:
+  it is demonstrated to scale and it's battle tested. It has been optimized during the years
+  to reduce dataset sizes on disk and to deliver excellent runtime performance.
+  It allows to access only part of the columns of the datasets, too.
+  The TNtuple and TNtupleD classes are specialisations of the TTree class which can store
```
Comment (pcanal) [inline tree/tree/src/TTree.cxx]:
```suggestion
  The TNtuple and TNtupleD classes are specialisations of the TTree class which can
```

File: tree/tree/src/TTree.cxx [hunk 2]
```diff
@@ -11,50 +11,55 @@
...
+  to only hold single precision and double precision floating-point numbers respectively;
```
Comment (pcanal) [inline tree/tree/src/TTree.cxx]:
```suggestion
  only hold single precision and double precision floating-point numbers respectively;
```

File: tree/tree/src/TTree.cxx [hunk 3]
```diff
@@ -11,50 +11,55 @@
...
 /** \class TTree
 \ingroup tree
 
-A TTree object has a header with a name and a title.
+A TTree represents a columnar dataset. Any C++ type can be stored in its columns.
 
-It consists of a list of independent branches (TBranch). Each branch has its own
-definition and list of buffers. Branch buffers may be automatically written to
-disk or kept in memory until the Tree attribute `fMaxVirtualSize` is reached.
+A TTree, often called in jeargon *tree*, consists of a list of independent columns or *branches*,
```
Comment (pcanal) [inline tree/tree/src/TTree.cxx]:
```suggestion
A TTree, often called in jargon *tree*, consists of a list of independent columns or *branches*,
```

**Comment by dpiparo** [comment]
Thanks Philippe!

**Comment by pcanal** [comment]
Thanks for this nice update :)

**Pull request merged**